ftd/src/interpreter/things/component.rs

```diff
@@ -1107,15 +1108,22 @@ fn search_things_for_module(
 pub struct Loop {
     pub on: ftd::interpreter::PropertyValue,
     pub alias: String,
+    pub key_name: Option<String>,
     pub line_number: usize,
 }
 
 impl Loop {
-    fn new(on: ftd::interpreter::PropertyValue, alias: &str, line_number: usize) -> Loop {
+    fn new(
+        on: ftd::interpreter::PropertyValue,
+        alias: &str,
+        key_name: Option<String>,
```

Comment:
Similarly refactor this variable name as well
```suggestion
        loop_counter_alias: Option<String>,
```